File: 921. Minimum Add to Make Parentheses Valid/_921.py
class Solution:
    def minAddToMakeValid(self, S: str) -> int:
        res = list()
        for i in S:
            if i == '(':
                res.append('(')
            else:
                if len(res) == 0:
                    res.append(')')
                elif res[-1] == '(':
                    res.pop()
                else:
                    res.append(')')
        return len(res)

File: 921. Minimum Add to Make Parentheses Valid/921.cpp
class Solution {
public:
    int minAddToMakeValid(string S) {
        vector<char> res;
        for(int i=0; i<S.length(); i++){
            if (S[i] == '('){
                res.push_back('(');
            }else{
                if (res.size() == 0){
                    res.push_back(')');
                }else if(res[res.size()-1] == '('){
                    res.pop_back();
                }else{
                    res.push_back(')');
                }
            }
        }
        return res.size();
    }
};